### ISSUE TITLE
Are the training loss and validation loss recorded?

### ISSUE THREAD

**Issue by shatealaboxiaowang** [ise-uiuc/magicoder#28]
Hi, Dear:

Thank you very much for your code. I am reproducing your training process. I wonder what your training loss and validation loss are during the training process, and I want to align them with your training process on dataset Magicoder-OSS-Instruct-75K and datasets--ise-uiuc--Magicoder-Evol-Instruct-110K

Thx

**Comment by shatealaboxiaowang** [comment]
Thank you for your open source, I am replicating your fine-tuning process according to the code on github. Do the results of train loss=0.16 and eval_loss=0.21 I trained on the 75k dataset match yours? I will continue training on the 110k dataset.
I trained for 4 epochs and indeed started overfitting after the second epoch.

**Comment by UniverseFly** [comment]
[Magicoder-S-CL.json](https://github.com/ise-uiuc/magicoder/files/14047808/Magicoder-S-CL.json)
[Magicoder-CL.json](https://github.com/ise-uiuc/magicoder/files/14047809/Magicoder-CL.json)
[Magicoder-S-DS.json](https://github.com/ise-uiuc/magicoder/files/14047810/Magicoder-S-DS.json)
[Magicoder-DS.json](https://github.com/ise-uiuc/magicoder/files/14047811/Magicoder-DS.json)

Hi, here are the trainer states. Hope they can help!

**Comment by shatealaboxiaowang** [comment]
> [Magicoder-S-CL.json](https://github.com/ise-uiuc/magicoder/files/14047808/Magicoder-S-CL.json) [Magicoder-CL.json](https://github.com/ise-uiuc/magicoder/files/14047809/Magicoder-CL.json) [Magicoder-S-DS.json](https://github.com/ise-uiuc/magicoder/files/14047810/Magicoder-S-DS.json) [Magicoder-DS.json](https://github.com/ise-uiuc/magicoder/files/14047811/Magicoder-DS.json)
> 
> Hi, here are the trainer states. Hope they can help!

Thank you very much, my training process is basically the same as your loss metric and the test results on humaneval dataset are basically consistent

**Comment by UniverseFly** [comment]
Good to hear you can reproduce it. Yeah we did observe that the infilling capability was at least not decreasing. We believe this is because the model learned some general alignment during the instruction tuning, and infilling is a kind of alignment based on the surrounding context. Further study of this phenomenon would be interesting.
